server/conn.go

```diff
@@ -1278,19 +1281,20 @@ func (cc *clientConn) handleQuery(ctx context.Context, sql string) (err error) {
 		return cc.writeOK()
 	}
 
+	warns := sc.GetWarnings()
+	parserWarns := warns[len(prevWarns):]
```

Comment:
so if we have warns in `warns[:len(prevWarns)]`, they also can not report to user too~?

Reply:
Those warnings are generated by previous statement, if the current query is `show warnings`, they would be reported, if not, they would be reset by `ResetContextOfStmt`.